Sources/NIOHTTP2/HTTP2ChannelHandler+InlineStreamMultiplexer.swift

```diff
@@ -255,7 +255,11 @@ extension NIOHTTP2Handler {
         ///       or ``HTTP2Frame/FramePayload`` if there are none.
         ///     - initializer: A callback that will be invoked to allow you to configure the
         ///         `ChannelPipeline` for the newly created channel.
+        @available(macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, *)
+        @_spi(AsyncChannel)
         public func createStreamChannel<Inbound, Outbound>(
+            backpressureStrategy: NIOAsyncSequenceProducerBackPressureStrategies.HighLowWatermark? = nil,
+            isOutboundHalfClosureEnabled: Bool = false,
             inboundType: Inbound.Type,
             outboundType: Outbound.Type,
```

Comment:
can we reorder these so `{in,out}boundType` come first? Required parameters should come first.

Reply:
We should add defaults to the types as well like this `= Inbound.self`. Then the order is the same as in our bootstrap

Reply:
I don't think we want to add defaults here to force people to consider what types are used. This was also mentioned in an earlier PR https://github.com/apple/swift-nio-http2/pull/403#discussion_r1250706741

Reply:
This is not really a default. If you write `inboundType: Inbound.Type = Inbound.self` it will allow the user to either write out the generic return type which will get inferred or pass parameters. If you don't add this default value then they always have to write out the parameters even though type inference doesn't need it.

Reply:
Yep, good point. I've added these in the latest commit and put the parameters back in the order to match `NIOAsyncChannel`.